### DISCUSSION TITLE
How to debug unit test?

### DISCUSSION THREAD

**Discussion by chunk1ty** [dotnet/runtime#69046] [Q&A]
When I run unit tests they passed (Microsoft.Extensions.Caching.Memory.Tests), but when I try to debug some of them it throws following exception:
![image](https://user-images.githubusercontent.com/6850746/167355390-370a1ed1-2e98-4a98-81ce-c94f47fce8fd.png)

Visual Studio 2022 17.1.6
Windows 11 22000.652

N.B: 
I followed these guides: 
https://github.com/dotnet/runtime/tree/main/docs/workflow https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/windows-requirements.md

Any idea why I am unable to debug unit tests?

**Comment by vitek-karas** [comment]
The exception might be expected by the test, or it might be caught by something before it gets to the test. Exception on its own doesn't necessarily mean something went wrong - especially in tests.

**Comment by chunk1ty** [comment]
[vitek-karas](https://github.com/vitek-karas) Its not expected by the test. I try to the debug followint test: SetAndGetReturnsObject()
https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Caching.Memory/tests/MemoryCacheSetAndRemoveTests.cs
I put a breakpoint then F11 and after a couple of minutes Visual Studio is being restarted.

**Comment by huoyaoyuan** [comment]
I remember I did `build.cmd libs.pretest -f net462` or something similar to get debugging work, otherwise it will complain about missing some test utility.

Edit: that will prevent building and starting at all.
